SPEC/DHT.md

```diff
@@ -51,6 +54,8 @@ If no `callback` is passed, a promise is returned.
 
 ```JavaScript
 ipfs.dht.findprovs(multihash, function (err, peerInfos) {})
+
+ipfs.dht.findprovs(multihash, { timeout: 4000 }, function (err, peerInfos) {})
```

Comment:
I'd like to add a test for this, but I'm not sure how to test this timeout.

Appreciate any points, in case anybody wants this as well :)

Reply:
I'd set the timeout really low and give it a hash that no-one will have, the callback should be called with an error before mocha times out the test.

Reply:
Okay, I'm looking into this now and - sorry if this is a dumb question - but I'm wondering how I should know what hash would be a hash that no one will have? I mean I can't really know this upfront right?

Or is it enough to know, that no one of the connected peers will have it in the testing scenario, which in this case would be `nodeB`, I guess? Considering this already existing testing setup:

```
common.setup((err, factory) => {
        expect(err).to.not.exist()

        spawnNodesWithId(2, factory, (err, nodes) => {
          expect(err).to.not.exist()

          nodeA = nodes[0]
          nodeB = nodes[1]

          connect(nodeB, nodeA.peerId.addresses[0], done)
        })
})
```

What I've tried now, was to create an empty dagNode object within `nodeB` to get hold of its CID, but I don't actually announce the content in the network with `provide()`:

```
it('should take options to override timout config', function (done) {
  const options = {
    timeout: 1
  }

  waterfall([
    (cb) => nodeB.object.new(cb),
    (dagNode, cb) => {
      // Notice I'm not announcing the hash in the network using `provide()` but simply return 
      // CID
      const cidV0 = new CID(dagNode.toJSON().multihash)
      cb(cidV0)
    },
    (cidV0, cb) => nodeA.dht.findprovs(cidV0, options, (err) => {
      // I'd expect this to be truthy as nodeA shouldn't find a provider for the given hash
      expect(err).to.exist()
      cb(err)
    }),
  ], done) // this fails due to non-error being passed to done
})

```
So this test fails because `done()` doesn't get called with an error object, which means that `findProvs` doesn't throw and error here. 

Can nodes find providers, even though other nodes didn't announce their content to the network?

Reply:
I think that by adding content via `object.new` that you become a provider for that CID automatically.

We have a helper for the webui tests that creates a CID from random bytes without adding content to IPFS. Perhaps something similar here might work?

https://github.com/ipfs-shipyard/ipfs-webui/blob/revamp/test/helpers/cid.js

Reply:
Yes, that could work. I'll give it a spin